shopify/product-reviews-template/extensions/product-reviews/blocks/productReviews.liquid

```diff
@@ -0,0 +1,41 @@
+<script src="{{ 'reviewStars.js' | asset_url }}" defer="defer"></script>
+{{ 'reviews.css' | asset_url | stylesheet_tag }}
+
+{% assign reviews = product.metafields.productReviews.reviewMetaobjects.value %}
+
+<script>
+    document.addEventListener("DOMContentLoaded", function () {
+      window.reviews = {{ reviews | json }};
+    });
+ </script>
+
+<h2>Reviews</h2>
+<div id="reviews">
+  {% for review in reviews%}
+    <div class="review-card">
+      <div class="review-content">
+        {{ review.content }}
+      </div>
+      <span class="vertical-divider">&nbsp;</span> 
+      <div class="review-stars" id="review-{{forloop.index0}}">
+        {% comment %} The stars will be rendered here using JS {% endcomment %}
```

Comment:
I kind of feel we should be doing this using Liquid. When you mentioned we were using JS, I assumed that we were also calling the Gadget API, but if we are loading with metafields, then maybe not

Reply:
I wasn't able to get the types for the metafield/metaobject rating value to match the value of the for loop index in Liquid so I decided to do it in JS.

Reply:
I had been breaking my head over it for days